core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -482,7 +489,13 @@ public boolean removeProperties(Namespace namespace, Set<String> properties)
 
   @Override
   public void close() {
-    connections.close();
+    if (closeableGroup != null) {
+      try {
```

Comment:
I don't think we need to wrap this in a try-catch block here and none of the other catalogs do, so I think we can remove this here

Reply:
This was mentioned above, no try-catch block will break existing APIs here.

Reply:
thanks for conforming, I forgot about that